prusti-encoder/src/encoders/type/predicate.rs

```diff
@@ -186,48 +173,111 @@ impl<'vir> PredicateEncOutputRef<'vir> {
         vid.map(|vid| self.expect_variant(vid).predicate)
             .unwrap_or(self.ref_to_pred)
     }
-    #[track_caller]
-    pub fn expect_variant_opt(
+
+    pub fn get_ref_to_pred(
         &self,
         vid: Option<abi::VariantIdx>,
-    ) -> &PredicateEncDataStruct<'vir> {
+    ) -> PredicateIdn<'vir, (vir::Ref, vir::ManyTyVal)> {
+        vid.map(|vid| {
+                let data = self.expect_enumlike().expect("empty enum");
+                data.variants[vid.as_usize()].predicate
+        }).unwrap_or(self.ref_to_pred)
+    }
+
+    /// Returns `None` if the `vid` is `None` and this is an enum.
+    pub fn get_variant_opt(
+        &self,
+        vid: Option<abi::VariantIdx>,
+    ) -> Option<&PredicateEncDataStruct<'vir>> {
         match vid {
```

Comment:
(Can this not be a `map` like in `get_ref_to_pred` as well?)

Reply:
We do explicitly want different behaviour in the two different cases though here, it's hard to use a map here